src/helpers/a11yAPI.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import makeQuery from './makeQuery';
+
+type TextQueryFn = (string | RegExp) => ReactTestInstance;
+type ArrayQueryFn = (string | Array<string>) => ReactTestInstance;
+
+type A11yAPI = {
+  getByA11yLabel: TextQueryFn,
+  getAllByA11yLabel: TextQueryFn,
+  queryByA11yLabel: TextQueryFn,
+  queryAllByA11yLabel: TextQueryFn,
+  getByA11yHint: TextQueryFn,
+  getAllByA11yHint: TextQueryFn,
+  queryByA11yHint: TextQueryFn,
+  queryAllByA11yHint: TextQueryFn,
+  getByA11yRole: TextQueryFn,
+  getAllByA11yRole: TextQueryFn,
+  queryByA11yRole: TextQueryFn,
+  queryAllByA11yRole: TextQueryFn,
+  getByA11yStates: ArrayQueryFn,
+  getAllByA11yStates: ArrayQueryFn,
+  queryByA11yStates: ArrayQueryFn,
+  queryAllByA11yStates: ArrayQueryFn,
+};
+
+export function matchStringValue(prop?: string, matcher: string | RegExp) {
+  if (!prop) {
+    return false;
+  }
+
+  if (typeof matcher === 'string') {
+    return prop === matcher;
+  }
+
+  return Boolean(prop.match(matcher));
+}
+
+export function matchArrayValue(
+  prop?: Array<string>,
+  matcher: string | Array<string>
+) {
+  if (!prop) {
+    return false;
+  }
+
+  if (typeof matcher === 'string') {
+    return prop.includes(matcher);
+  }
+
+  // $FlowFixMe
+  return !matcher.some(e => !prop.includes(e));
+}
+
+const a11yAPI = (instance: ReactTestInstance): A11yAPI =>
+  ({
+    ...makeQuery(
+      'accessibilityLabel',
+      {
+        getBy: 'getByA11yLabel',
```

Comment:
We could also replace `strings` with `Array<string>` to have more than one alias for a query.
E.g. `getByA11yLabel` and `getByAccessibilityLabel`. @thymikee wdyt?

Reply:
I'm down 👍